src/Client/ZoomSyncClient.php

```diff
@@ -0,0 +1,297 @@
+<?php
+
+namespace Drupal\openy_gc_zoom_sync\Client;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\openy_gc_zoom_sync\ZoomSyncClientInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * ZoomSync client.
+ */
+class ZoomSyncClient implements ZoomSyncClientInterface {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * An http client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * Psr\Log\LoggerInterface definition.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Zoom API Secret.
+   *
+   * @var string
+   */
+  protected $apiSecret;
+
+  /**
+   * Zoom API base url.
+   *
+   * @var string
+   */
+  protected $baseUri;
+
+  /**
+   * ZoomSyncClient constructor.
+   *
+   * @param \Drupal\Core\Config\ImmutableConfig $config
+   *   Config.
+   */
+  public function __construct(
+    ClientInterface $httpClient,
+    ConfigFactoryInterface $config_factory,
+    LoggerInterface $logger) {
+
+    $this->httpClient = $httpClient;
+    $this->logger = $logger;
+    $this->configFactory = $config_factory->get('zoom_sync.settings');
+    $this->apiSecret = $this->configFactory->get('data_service_token');
+    $this->baseUri = $this->configFactory->get('data_service_url');
+  }
+
+  /**
+   * Utilizes Drupal's httpClient to connect to the Zoom API.
+   *
+   * Info: https://marketplace.zoom.us/docs/api-reference/introduction
+   * Currently just supports JWT authentication.
+   *
+   * @param string $method
+   *   get, post, patch, delete, etc. See Guzzle documentation.
+   * @param string $endpoint
+   *   The Zoom API endpoint (ex. users)
+   * @param array $query
+   *   Any Query Parameters defined in the API spec.
+   * @param array $body
+   *   Array that will get converted to JSON for some requests.
+   *
+   * @return mixed
+   *   RequestException or \GuzzleHttp\Psr7\Response body
+   */
+  public function request(string $method, string $endpoint, array $query = [], array $body = []) {
+    try {
+      $response = $this->httpClient->{$method}(
+        $this->baseUri . $endpoint,
+        $this->buildOptions($query, $body)
+      );
+      // TODO: Add additional response options.
+      $payload = Json::decode($response->getBody()->getContents());
+      return $payload;
+    }
+    catch (RequestException $exception) {
+      // Log Any exceptions.
+      $this->logger->error('Failed to complete Zoom API Task "%error"', [
```

Comment:
Let's answer to "so what?" question with action item for the user
Like - please contact support by calling: 123456678, or - "check Zoom settings at "link to form"